File: api/schemas/scanner.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class UniverseOption(BaseModel):
    id: str
    label: str
    kind: str
    count: int | None = None


class ScanRequest(BaseModel):
    universe: str = Field(default="demo_sample.csv")
    mode: Literal["live", "sample"] = "sample"


class ScanResultRow(BaseModel):
    ticker: str
    score: int
    gap_pct: float | None = None
    rel_dollar_vol: float | None = None
    avg20_dollar_vol: float | None = None
    price: float | None = None
    premarket_last: float | None = None
    reasons: list[str] = Field(default_factory=list)


class ScanResponse(BaseModel):
    universe: str
    mode: str
    row_count: int
    columns: list[str]
    results: list[ScanResultRow]


class HealthResponse(BaseModel):
    status: str
    app: str

File: web/types/scanner.ts
export type ScanMode = "live" | "sample";

export interface UniverseOption {
  id: string;
  label: string;
  kind: string;
  count?: number;
}

export interface ScanResultRow {
  ticker: string;
  score: number;
  gap_pct: number | null;
  rel_dollar_vol: number | null;
  avg20_dollar_vol: number | null;
  price: number | null;
  premarket_last: number | null;
  reasons: string[];
}

export interface ScanResponse {
  universe: string;
  mode: ScanMode;
  row_count: number;
  columns: string[];
  results: ScanResultRow[];
}
